### PR TITLE
Convert relative links to hardlinks

### PR DESCRIPTION
**Pull request by Rafase282**
This is the final edit to get all relative links working properly. It converts them to hardlink using regex.

@benmcmahon100 

closes https://github.com/FreeCodeCamp/wiki/issues/558
closes FreeCodeCamp/camperbot#64

### CODE REVIEW

File: lib/utils/TextLib.js [hunk 1]
```diff
@@ -9,8 +9,21 @@ const TextLib = {
 	if(typeof baseUrl === "undefined"){
 		baseUrl = 'http://github.com/FreeCodeCamp/FreeCodeCamp/wiki/';
 	}
-    const linkMatchRegExp = /\[.+\]\((.+)\)/g;
-    return wikiContent.replace(linkMatchRegExp, baseUrl + '$1');
+    const linkMatchRegExp = /.+\]\((.+)\)/g;
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: reformat source code with proper indentation

File: lib/bot/KBase.js [hunk 2]
```diff
@@ -110,28 +109,29 @@ const KBase = {
     Utils.log('shortList', shortList);
 
     const emojiList = [':zero:', ':one:', ':two:', ':three:', ':four:',
```
Comment (abhisekp) [inline lib/bot/KBase.js]:
```js
    const emojiList = [
      ':zero:', ':one:', ':two:', ':three:', ':four:',
      ':five:', ':six:', ':seven:', ':eight:', ':nine:'
    ];
```

File: lib/bot/KBase.js [hunk 3]
```diff
@@ -148,4 +148,4 @@ const KBase = {
 
 KBase.initSync();
 
-module.exports = KBase;
+module.exports = KBase;
```
Comment (abhisekp) [inline lib/bot/KBase.js]: newline at the end

File: lib/utils/TextLib.js [hunk 4]
```diff
@@ -39,4 +54,4 @@ const TextLib = {
   }
 };
 
-module.exports = TextLib;
+module.exports = TextLib;
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: add newline at the end

File: lib/utils/TextLib.js [hunk 5]
```diff
@@ -5,14 +5,27 @@ const AppConfig = require('../../config/AppConfig');
...
   // we only show the first para
   // and limit to 20 lines
-  fixRelativeLink: function fixRelativeLink(wikiContent, baseUrl ) {
-	if(typeof baseUrl === "undefined"){
-		baseUrl = 'http://github.com/FreeCodeCamp/FreeCodeCamp/wiki/';
-	}
-    const linkMatchRegExp = /\[.+\]\((.+)\)/g;
-    return wikiContent.replace(linkMatchRegExp, baseUrl + '$1');
+  fixRelativeLink: function fixRelativeLink(wikiContent, baseUrl) {
+    if (typeof baseUrl === "undefined") {
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: strings must have single quotes

File: lib/utils/TextLib.js [hunk 6]
```diff
@@ -5,14 +5,27 @@ const AppConfig = require('../../config/AppConfig');
...
+      baseUrl = 'http://github.com/FreeCodeCamp/FreeCodeCamp/wiki/';
+    }
+    const linkMatchRegExp = /.+\]\((.+)\)/g;
+    return wikiContent.split('\n').map((line) => {
+      if (line.match(linkMatchRegExp) && line.match(linkMatchRegExp).length > 0) {
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: Less than 80 chars per line

```js
      if (line.match(linkMatchRegExp) &&
          line.match(linkMatchRegExp).length > 0) {
```

File: lib/utils/TextLib.js [hunk 7]
```diff
@@ -5,14 +5,27 @@ const AppConfig = require('../../config/AppConfig');
...
+          if (word.match(linkMatchRegExp)) {
+            if (!word.match(/\:\/\//gi)) {
+              word = word.replace(/\((.+)\)/gi, "(" + baseUrl + '$1' + ")");
+              return word;
+            }
+          }
+          return word;
+        }).join(' ');
+      }
+      return (line)
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: - missing semicolon
- remote parens `( ... )`

**Comment by abhisekp** [comment]
Relative links are not working properly.
try **`bot python res`** command.

File: lib/bot/KBase.js [hunk 8]
```diff
@@ -1,9 +1,9 @@
 'use strict';
 
 const TextLib = require('../utils/TextLib'),
```
Comment (abhisekp) [inline lib/bot/KBase.js]: Pretty indentation

```js
const TextLib = require('../utils/TextLib'),
      fs = require('fs'),
      path = require('path'),
      Utils = require('../utils/Utils');
```

**Comment by Rafase282** [comment]
Made changes, counted characters 80 or less per line.

File: lib/utils/TextLib.js (hunk 6 above)
Comment (ltegman) [inline lib/utils/TextLib.js]: @Rafase282 This line is still too long.

File: lib/utils/TextLib.js [hunk 9]
```diff
@@ -5,14 +5,32 @@ const AppConfig = require('../../config/AppConfig');
...
+    }
+    const linkMatchRegExp = /.+\]\((.+)\)/g;
+    return wikiContent.split('\n').map((line) => {
+      if (line.match(linkMatchRegExp) &&
+          line.match(linkMatchRegExp).length > 0) {
+        line = line.split(' ').map((word) => {
+          if (word.match(linkMatchRegExp)) {
+            if (!word.match(/\:\/\//gi)) {
+              if (word.match(/\(\#(.+)/gi)){
+                word = word.replace(/\(\#(.+)/gi, '(' + baseUrl + topicName + '#' + '$1');
```
Comment (abhisekp) [inline lib/utils/TextLib.js]: use string templates

**Comment by abhisekp** [comment]
# :+1: LGTM
[](https://i.gyazo.com/ec4af70d8021940213595481b920d72b.png)

**Pull request merged**